docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,225 @@
+# ORAS CLI Error Handling and Message Guidelines
+
+This document aims to provide the guidelines for ORAS contributors to improve existing error messages and error handling method as well as the new error output format. It will also provide recommendations for ORAS CLI contributors for how to write friendly and standard error messages and avoid inconsistent error handling & messages.
+
+## General guiding principles
+
+A clear and actionable error message is very important when raising an error, so make sure your error message describes clearly what the error is and tells users what they need to do if possible.
+
+First and foremost, make the error messages descriptive and informative. Error messages are expected to be helpful to troubleshoot where the user has done something wrong and the program is guiding them in the right direction. A great error message is recommended to contain the following:
+
+- Error code
+- Error title
+- Error description 
+- How to fix the error
+- URL for more information (Optional, it can be a TSG document link)
```

Comment:
We can put a guideline on using a URL that it will not likely be changed. We met this problem in Notation that the URL changed.

Reply:
Yes, That's being said we should use a versioned document link in error message